server/tserver/src/main/java/org/apache/accumulo/tserver/util/CreateEmpty.java

```diff
@@ -0,0 +1,200 @@
+  public String description() {
+    return "Creates empty rfiles and empty wal files for emergency recovery";
+  }
+
+  @Override
+  public void execute(String[] args) throws Exception {
+
+    Opts opts = new Opts();
+    opts.parseArgs("accumulo create-empty", args);
+
+    var siteConfig = opts.getSiteConfiguration();
+    try (ServerContext context = new ServerContext(siteConfig)) {
+      switch (opts.fileType) {
+        case rfile:
+          createEmptyRFile(opts, context);
+          break;
+        case wal:
+          createEmptyWal(opts, context);
+          break;
+        default:
+          throw new ParameterException(
+              "file type must be rfile or wal, received: " + opts.fileType);
+      }
+    }
+  }
+
+  void createEmptyRFile(final Opts opts, final ServerContext context) throws IOException {
+    var vm = context.getVolumeManager();
+
+    CryptoEnvironment env = new CryptoEnvironmentImpl(CryptoEnvironment.Scope.TABLE);
+    CryptoService cryptoService = context.getCryptoFactory().getService(env,
+        context.getConfiguration().getAllCryptoProperties());
+
+    for (String filename : opts.files) {
+      Path path = new Path(filename);
+      checkFileExists(path, vm);
+      UnreferencedTabletFile tabletFile =
+          UnreferencedTabletFile.of(vm.getFileSystemByPath(path), path);
+      LOG.info("Writing to file '{}'", tabletFile);
+      FileSKVWriter writer = new RFileOperations().newWriterBuilder()
+          .forFile(tabletFile, vm.getFileSystemByPath(path), context.getHadoopConf(), cryptoService)
+          .withTableConfiguration(DefaultConfiguration.getInstance()).withCompression(opts.codec)
+          .build();
+      writer.close();
+    }
+  }
+
+  void createEmptyWal(Opts opts, ServerContext context) throws IOException {
+    final LogFileValue EMPTY = new LogFileValue();
+
+    var vm = context.getVolumeManager();
+
+    for (String filename : opts.files) {
+      Path path = new Path(filename);
+      checkFileExists(path, vm);
+      try (var out = new DataOutputStream(vm.create(path))) {
+        LOG.info("Output file: {}", path);
+
+        out.write(LOG_FILE_HEADER_V4.getBytes(UTF_8));
+
+        CryptoEnvironment env = new CryptoEnvironmentImpl(CryptoEnvironment.Scope.WAL);
+        CryptoService cryptoService = context.getCryptoFactory().getService(env,
+            context.getConfiguration().getAllCryptoProperties());
+
+        byte[] cryptoParams = cryptoService.getFileEncrypter(env).getDecryptionParameters();
+        CryptoUtils.writeParams(cryptoParams, out);
+
+        LogFileKey key = new LogFileKey();
+        key.event = OPEN;
+        key.tserverSession = "";
+
+        key.write(out);
+        EMPTY.write(out);
+      }
+    }
+  }
+
+  private void checkFileExists(final Path path, final VolumeManager vm) throws IOException {
+    if (vm.exists(path)) {
+      throw new IllegalArgumentException(
+          "file: '" + path.getParent() + "/" + path.getName() + "' exists");
```

Comment:
The `file:` prefix could be confusing for local filesystem URLs

```suggestion
          path.getParent() + "/" + path.getName() + " exists");
```